src/Features/CSharp/Portable/CodeRefactorings/InvertIf/InvertIfCodeRefactoringProvider.cs

```diff
@@ -1,93 +1,24 @@
-            {
-                return null;
-            }
-
-            var span = TextSpan.FromBounds(ifStatement.IfKeyword.Span.Start, ifStatement.CloseParenToken.Span.End);
-            if (!span.IntersectsWith(textSpan))
-            {
-                return null;
-            }
-
-            return ifStatement;
+            return token.GetAncestor<IfStatementSyntax>();
         }
 
-        protected override SyntaxNode GetRootWithInvertIfStatement(
-            Document document,
-            SemanticModel model,
-            SyntaxNode ifStatementSyntax,
-            CancellationToken cancellationToken)
+        protected override IAnalyzer GetAnalyzer(SyntaxNode ifStatement)
         {
-            var generator = SyntaxGenerator.GetGenerator(document);
-            var syntaxFacts = GetSyntaxFactsService();
-
-            var oldIfStatement = (IfStatementSyntax)ifStatementSyntax;
-
-            // For single line statement, we swap the TrailingTrivia to preserve the single line
-            StatementSyntax newIfNodeStatement = null;
-            ElseClauseSyntax newElseStatement = null;
-
-            var hasNewLineAfterClosingBrace = oldIfStatement.Statement.GetTrailingTrivia().Any(trivia => trivia.Kind() == SyntaxKind.EndOfLineTrivia);
-            if (hasNewLineAfterClosingBrace)
-            {
-                newIfNodeStatement = oldIfStatement.Else.Statement.Kind() != SyntaxKind.Block
-                    ? SyntaxFactory.Block(oldIfStatement.Else.Statement)
-                    : oldIfStatement.Else.Statement;
-                newElseStatement = oldIfStatement.Else.WithStatement(oldIfStatement.Statement);
-            }
-            else
-            {
-                var elseTrailingTrivia = oldIfStatement.Else.GetTrailingTrivia();
-                var ifTrailingTrivia = oldIfStatement.Statement.GetTrailingTrivia();
-                newIfNodeStatement = oldIfStatement.Else.Statement.WithTrailingTrivia(ifTrailingTrivia);
-                newElseStatement = oldIfStatement.Else.WithStatement(oldIfStatement.Statement).WithTrailingTrivia(elseTrailingTrivia);
-            }
-
-            var newIfStatment = oldIfStatement.Else.Statement.Kind() == SyntaxKind.IfStatement && newIfNodeStatement.Kind() != SyntaxKind.Block
-                ? SyntaxFactory.Block(newIfNodeStatement)
-                : newIfNodeStatement;
-
-            oldIfStatement = oldIfStatement.WithCondition((ExpressionSyntax)(Negate(oldIfStatement.Condition, generator, syntaxFacts, model, cancellationToken)))
-                .WithStatement(newIfStatment)
-                .WithElse(newElseStatement);
-
-            if (hasNewLineAfterClosingBrace)
-            {
-                oldIfStatement = oldIfStatement.WithAdditionalAnnotations(Formatter.Annotation);
-            }
-
-            // get new root
-            return model.SyntaxTree.GetRoot(cancellationToken).ReplaceNode(ifStatementSyntax, oldIfStatement);
+            return new Analyzer();
```

Comment:
consider making singleton.  you don't need a fresh instance it seems.

Reply:
VB did this to me. (see the VB overrides)

Reply:
(resolved)